daemon/logs.go

```diff
@@ -19,10 +20,38 @@ import (
 	"github.com/docker/docker/pkg/stdcopy"
 )
 
+type loggerStore struct {
```

Comment:
I probably don't understand the caching part but by can't we fix a logger to a container on container creation and remove the id based accessing functions from daemon package.